### PR TITLE
Drop python 3.8 and numpy 1.20

### PR DESCRIPTION
**Pull request by joshmoore**
Various PR builds (e.g. https://github.com/zarr-developers/zarr-python/actions/runs/6704243768/job/18216294128?pr=1543) are failing on Python 3.8. According to https://scientific-python.org/specs/spec-0000/ dropping Python **3.9** is suggested by the end of this year. As a step in that direction, this PR drops 3.8 ~~and adds 3.12~~.

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

**Comment by joshmoore** [comment]
Note: green builds will require updating the GitHub settings:

image

**Comment by d-v-b** [comment]
this is great idea, thank you

**Review by d-v-b — APPROVED** [review]

**Comment by joshmoore** [comment]
May have been a bit premature with 3.12...

```
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for line_profiler
```

**Comment by joshmoore** [comment]
All non-3.8 builds have passed. I think this is in a fine state. @jakirkham, any thoughts, after you [recently](https://github.com/zarr-developers/numcodecs/pull/471) added 3.12 to numcodecs but didn't drop 3.8?

**Comment by jhamman** [comment]
+1 on dropping 3.8 then 3.9 early next year. @joshmoore - can we help you get this over the line?

File: .github/workflows/releases.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
       - uses: actions/setup-python@v4.7.1
         name: Install Python
         with:
-          python-version: '3.8'
+          python-version: '3.9'
```
Comment (jhamman) [inline .github/workflows/releases.yml:22]: I lean toward publishing with the latest python release that we support.
```suggestion
          python-version: '3.11'
```

**Review by jhamman — APPROVED** [review]

**Comment by joshmoore** [comment]
I accepted your change, @jhamman, and merged in main. Assuming all is green, I imagine we can get this in immediately (unless we want one last pre-drop release).

**Comment by jhamman** [comment]
> I accepted your change, @jhamman, and merged in main. Assuming all is green, I imagine we can get this in immediately (unless we want one last pre-drop release).

I don't think that is necessary. Let's merge when this goes green!

**Comment by jhamman** [comment]
@joshmoore - any idea why the last few checks have not reported back?

**Review by jakirkham — APPROVED** [review]

**Comment by joshmoore** [comment]
Got a (big) verbal :+1: from @jakirkham. The failing jobs are due to the job configuration re: https://github.com/zarr-developers/zarr-python/pull/1557#issuecomment-1786816094

I'll merge now and bump the 3.8s to 3.9s.

**Comment by joshmoore** [comment]
For now I've bumped the Windows and just removed the `build` items since the new versions hadn't been built yet:

image

**Pull request merged**